File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        
        // Using Hashing
        map<int, int> mp;
        for(int num: nums){
            mp[num]++;
        }

        int major=0;
        int n = nums.size();

        for(auto it : mp){
            if(it.second > n/2){
                return it.first;
            }
        }

        return -1;
    }
};

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        // O(n) time complexicity
        int element = 0, count=0;

        for(int i=0; i<nums.length; i++){
            if(count == 0){
                element = nums[i];
                count++;
            }
            else if(nums[i] == element){
                count++;
            }else{
                count--;
            }
        }

        return element;
    }
}